internal/tools/api_resources.go

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package tools
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/asgardeo/go/pkg/api_resource"
+	"github.com/asgardeo/mcp/internal/asgardeo"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+)
+
+func GetListAPIResourcesTool() (mcp.Tool, server.ToolHandlerFunc) {
+	client, err := asgardeo.GetClientInstance(context.Background())
+	if err != nil {
+		log.Printf("Error initializing client instance: %v", err)
+	}
+
+	apiResourceListTool := mcp.NewTool("list_api_resources",
+		mcp.WithDescription("List API Resources registered in Asgardeo"),
+		mcp.WithString("filter",
+			mcp.Description(`Filter expression to apply, e.g., name eq Payments API`),
+		),
+		mcp.WithString("before",
+			mcp.Description(`The before cursor to use for pagination. The API will return results before this cursor`),
+		),
+		mcp.WithString("after",
+			mcp.Description(`The after cursor to use for pagination. The API will return results after this cursor`),
+		),
+		mcp.WithNumber("limit",
+			mcp.Description(`The maximum number of results to return.`),
+		),
+	)
+
+	var apiResourceListToolImpl server.ToolHandlerFunc
+	apiResourceListToolImpl = func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
```

Comment:
We can use this syntax, can't we?

```suggestion
	apiResourceListToolImpl := func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
```

Reply:
Fixed with https://github.com/asgardeo/asgardeo-mcp-server/pull/7/commits/5b8a0b0a2af21b8443ae1ad324f416229caa39a8